llvm/include/llvm/SYCLLowerIR/LowerWGLocalMemory.h

```diff
@@ -0,0 +1,33 @@
+//===-- LowerWGLocalMemory.h - SYCL kernel local memory allocation pass ---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Replaces calls to __sycl_allocateLocalMemory(Size, Alignment) function with
```

Comment:
The pass does more - adding new globals. This is also part of key actions the pass does, please describe shortly - what they model, that it is 1-per allocation, how IR changes.

Reply:
Thank you very much for a thorough review and helpful hints.
Modified pass description in the header.